mbed_tools_ci/utils/package_helpers.py

```diff
@@ -0,0 +1,178 @@
+"""Utilities for retrieving Python's package information."""
+
+import logging
+import re
+from typing import List, Any, Optional
+
+import pkg_resources
+
+from .configuration import ConfigurationVariable, configuration
+from .definitions import UNKNOWN
+
+logger = logging.getLogger(__name__)
+
+
+class PackageMetadata:
+    """Package's metadata.
```

Comment:
Based on this description, I'm not sure when I would want to use this. Why is it needed?

Reply:
All the data about the licence, the version, the author, the name of a package is retrieved from the Python package metadata and this is a way to fetch it from Python packages